lib/utils.js

```diff
@@ -0,0 +1,58 @@
+// @flow
+
+import {Range} from 'atom';
+
+export default class Utils {
+  /**
+   * Obtain the range of the word at the given editor position.
+   * Uses the non-word characters from the position's grammar scope.
+   */
+  static getWordAtPosition(editor: TextEditor, position: atom$Point): Range {
+    const scopeDescriptor = editor.scopeDescriptorForBufferPosition(position);
+    const nonWordCharacters = Utils.escapeRegExp(
+      editor.getNonWordCharacters(scopeDescriptor.getScopesArray()),
```

Comment:
This accepts a `ScopeDescriptor` directly, no need to pull out the array. (It _is_ non-public though.) (It [ends up](https://github.com/atom/atom/blob/v1.16.0/src/text-editor-registry.js#L399) being a simple [`atom.config.get`](https://atom.io/docs/api/v1.16.0/Config#instance-get).)